### PR TITLE
Merge class LinearGetValues

### LINKED ISSUE #21: Implement the LinearGetValues class
Similar to `GetValues` except it operates on `Increment` and so performs the tangent linear / adjoint for interpolation.
`TestLinearGetValues` should pass after this

`GetValues` is probably easier, so that one should be done first.

### PR DESCRIPTION
**Pull request by loganchen39**
@travissluka passed all the tests. Please check the lineargetvalues.yml where I didn't add anything under "linear getvalues test: ", not sure what to add. Let me know if it's ok. 

close #21.

### CODE REVIEW

File: src/umdsst/GetValues/CMakeLists.txt [hunk 1]
```diff
@@ -10,4 +10,6 @@ umdsst_target_sources(
     LocationsWrapper.f90
     GeoVaLsWrapper.h
     GeoVaLsWrapper.f90
+    GeoVaLsWrapperAD.h
+    GeoVaLsWrapperAD.f90
```
Comment (travissluka) [inline src/umdsst/GetValues/CMakeLists.txt:14]: Delete these, I don't see why they're needed. Just use  `GeoVaLsWrapper`, they're essentially the same.

**Review by travissluka — CHANGES_REQUESTED** [review]
Only small comment is to delete the `GeoVaLsWrapperAD` class as it isn't needed.

That's fine about no parameters being added to the yaml section, it seems the default values for the test are acceptable.

File: src/umdsst/GetValues/CMakeLists.txt (hunk 1 above)
Comment (loganchen39) [reply src/umdsst/GetValues/CMakeLists.txt:14]: @travissluka My understanding of the difference is that, in GeoVaLsWrapper.f90, we copy data from interpolated results to geovals as indicated below at line 57:
geovals%geovals(ivar)%vals(:,i) = field_data(:,i)

while for function void LinearGetValues::fillGeoVaLsTL(...), we need to copy the data from geovals as indicated below at line 52 within file GeoVaLsWrapperAD.f90
field_data(:,i) = geovals%geovals(ivar)%vals(:,i)
and then do the backward interpolation.

Is my understanding correct? If so, are you suggesting that I combine these two (files and classes) together? That makes sense.

Comment (travissluka) [reply src/umdsst/GetValues/CMakeLists.txt:14]: you're absolutely correct! Sorry, I'm not sure what I was thinking (been a long week??)
Those wrapper classes will be deleted eventually anyway once GeoVaLs is refactored by the JEDI team.

Looks good. I'll merge

**Review by travissluka — APPROVED** [review]

**Pull request merged**